js/Model/Project/ProjectStorage.js

```diff
@@ -0,0 +1,121 @@
+var SPM = SPM || {};
+(function() {
+
+SPM.Model = SPM.Model || {};
+SPM.Model.Project = SPM.Model.Project || {};
+
+var _projects = {};
+var _projectsByChannel = {};
+var _projectsByUser = {};
+var _projectsBySearch = {};
+var _me = null;
+var _utils = null;
+
+SPM.Model.Project.ProjectStorage = {
+    setMe: function(me) {
+        _me = me;
+    },
+
+    setUtils: function(utils) {
+        _utils = utils;
+    },
+
+	saveProject: function(project) {
+		_projects[project.id] = project;
+		if (project.slack) {
+			_projectsByChannel[project.slack] = project.id;
+		}
+        if (project.members) {
+            _.map(project.members, function(member) {
+                if (typeof _projectsByUser[member.id] == 'undefined') {
```

Comment:
Nop. If we do that, then a call to getProjectByUser / getMyProject might return a partial list of project instead of no data.